dags/vm_resource.py

```diff
@@ -157,25 +158,86 @@ class RuntimeVersion(enum.Enum):
   V2_ALPHA_TPUV5 = "v2-alpha-tpuv5"
 
 
-class ClusterName(enum.Enum):
-  """Common XPK cluster names."""
-
-  V4_8_CLUSTER = "mas-v4-8"
-  V4_32_CLUSTER = "mas-v4-32"
-  V5E_4_CLUSTER = "mas-v5e-4"
-  V5E_16_CLUSTER = "mas-v5e-16"
-  V4_8_MULTISLICE_CLUSTER = "v4-8-maxtext"
-  V4_16_MULTISLICE_CLUSTER = "v4-16-maxtext"
-  V4_128_MULTISLICE_CLUSTER = "v4-128-bodaborg-us-central2-b"
-  V5P_8_MULTISLICE_CLUSTER = "v5p-8-bodaborg-us-east5-a"
-  V5E_16_MULTISLICE_CLUSTER = "v5e-16-bodaborg"
-  V5E_256_MULTISLICE_CLUSTER = "v5e-256-bodaborg"
-  V5E_256_US_WEST_4_MULTISLICE_CLUSTER = "v5e-256-bodaborg-us-west4"
-  BODABORG_V6E_256 = "bodaborg-v6e-256"
-  A3_CLUSTER = "maxtext-a3-20n"
-  A3PLUS_CLUSTER = "a3plus-benchmark"
-  CPU_M1_MEGAMEM_96 = "m1-megamem-96-shared"
-  CPU_N2_STANDARD_64 = "shared-n2-standard-64"
+class XpkClusters:
+  """Common XPK cluster configs."""
+
+  V4_8_CLUSTER = XpkClusterConfig(
```

Comment:
Thanks Jon! This is great! 

I see V4_8_CLUSTER and BODABORG_V6E_256. Wondering if it's better we just capitalize the cluster name itself as variable name? I am not sure how frequent we will update the cluster name and node numbers. Will it be possible single slice cluster become multislice in the future? or verse visa. 

I prefer one of following two options:
1) directly use cluster name here; or
2) format all of those in the format `<vm_type>_<vm_shape>_<optional identifier>_cluster`, for example TPU_V4_32_CLUSTER and GPU_A3_CLUSTER, CPU_M1_MEGAMEM_96_CLUSTER, etc. optional identifier could be single slice, multislice, etc.

How do you think?

Reply:
Synced up offline - we'll use the cluster name here.